Examples/Algorithms/TrackFittingChi2/include/ActsExamples/TrackFittingChi2/TrackFittingChi2Algorithm.hpp

```diff
@@ -0,0 +1,130 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2019-2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/TrackFitting/Chi2Fitter.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/Measurement.hpp"
+#include "ActsExamples/EventData/Track.hpp"
+#include "ActsExamples/Framework/BareAlgorithm.hpp"
+#include "ActsExamples/MagneticField/MagneticField.hpp"
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+namespace Acts {
+class TrackingGeometry;
+}
+
+namespace ActsExamples {
+
+class TrackFittingChi2Algorithm final : public BareAlgorithm {
```

Comment:
Wouldn't it be possible to modify the track fitting algorithm we already have to make this possible? As far as I see there are not many differences. The `nUpdates` parameter could be either added to the normal `TrackFittingAlgorithm::Config` and then just be used in chi2 mode or to the parameters when constructing the fitter function.

Reply:
Good idea. I will put this on my list for future improvements, but for now I would prefer to have it strictly separeted from the other fitters.